ownCloud/Client/Actions/ClientDirectoryPickerViewController.swift

```diff
@@ -0,0 +1,97 @@
+//
+//  ClientDirectoryPickerViewController.swift
+//  ownCloud
+//
+//  Created by Pablo Carrascal on 22/08/2018.
+//  Copyright © 2018 ownCloud GmbH. All rights reserved.
+//
+
+import UIKit
+import ownCloudSDK
+
+class ClientDirectoryPickerViewController: ClientQueryViewController {
+
+	private var selectButton: ThemeButton
+	private var completion: (OCItem) -> Void
+	private var cancelBarButton: UIBarButtonItem!
+
+	init(core inCore: OCCore, query inQuery: OCQuery, completion: @escaping (OCItem) -> Void) {
+		selectButton = ThemeButton()
+		self.completion = completion
+		super.init(core: inCore, query: inQuery)
```

Comment:
Since this is a directory picker, it would be a nice touch to add a filter to the query, so it only provides directories. Right now, the picker also shows files and allows selecting them.

May also be worth considering creating the OCQuery in the VC, based on a path that's provided instead.